src/runtime/native/runtime.ml

```diff
@@ -114,11 +126,30 @@ let dump () =
         verbose Unable_to_write_file);
       close_out_noerr channel
 
+let sigterm_handler (_ : int) =
+  bisect_file_written := true;
+  dump ();
+  exit 0
+
+let dump_at_exit () =
```

Comment:
Last remaining issue. I think this function is now missing a `bisect_file_written := true`, which should probably come after the masking of the signal in the `!sigterm_enable` case.

Reply:
However, maybe I am misunderstanding something, because looking back at the new cram test, I would have expected 3 coverage files, one from the parent and two from the child, but the test shows the "proper" 2 files. Not sure what is going on.

Reply:
makes sense, fixed in the last version

Reply:
Ah, I believe we are not observing anything because the last function added with `at_exit` is the one added by the "user" program in the test, so it runs first, and SIGTERM comes during that function. That also means that the test isn't actually testing re-enabling of signals.

Reply:
The Bisect runtime's `at_exit` registration comes so early in the initialization of the program (since it is inserted at the start of every instrumented source file) that it would be difficult to make it run not last. I think you would have to do something like have a non-instrumented sub-library that the instrumented binary depends on, and the sub-library would have to do `at_exit`. That seems pretty excessive a burden for a test in a PR. I suggest to remove the current test, and I'll add the fancy early `at_exit` test later on.

Reply:
Or we can just leave the test and I'll fix it up later :) Anyway, I'm ready to merge now, let me know what you want to do or mark the PR as non-draft :)

Reply:
Ah, my brain is fried right now so I'll have to check tomorrow. I know at least that if I do not re-enable the signal, then the new cram test fails.

Reply:
> I think you would have to do something like have a non-instrumented sub-library that the instrumented binary depends on, and the sub-library would have to do at_exit.

that's what I've attempted to do, notice that `at_exit_hook.ml` is not instrumented.

Reply:
Ah. But AFAICT `at_exit_hook.ml` is still instrumented, because the `dune` file [emitted](https://github.com/aantron/bisect_ppx/pull/390/files#diff-ebcf58b941ad53f583173fdee96923dca6ff3ecbdb1f20ef8c85f13c56169361R19-R21) in the test still treats the whole set of `.ml` files as just an executable and the `preprocess` stanza is applying to all the `.ml` files pulled in during the dependency scan, since they get treated as part of the executable (there is no `library` stanza).

Ok, can wait for tomorrow. My brain is also a bit fried :) However, I also want to note that needing to fiddle a bit with a very edge case test doesn't block merging a PR, IMO :)

Reply:
Well I somehow missed the `[@@@coverage exclude_file]`! Sorry. Ok, at this point I am just confused about that test, which makes me all the more ready to merge this PR so I can do an investigation later, unless you strongly prefer to do it on your end :)

Reply:
Here is the order of how things run, as per a sprinkling of
`print_endline`:

```
+  at_exit_hook.ml: register at_exit (before fork)
+   *** (499322) register sigterm handler
+   *** (499322) register dump
+  at_exit_main.ml: parent: start
+  at_exit_main.ml: child: finished
+   *** (499486) in dump_at_exit
+  at_exit_hook.ml: child sleeps
+  at_exit_main.ml: parent: kill child
+  at_exit_main.ml: parent: finished
+   *** (499322) in dump_at_exit
```

so the top-level definitions of `at_exit_hook.ml` are evaluated first
(as it is referred to by `at_exit_main.ml`). this module does not
contain the bisect instrumentation (as can be verified with `$ ocamlc
-dsource ../../_build/default/test/sigterm/at_exit_hook.pp.ml`).  )

so all it does is register the applications at_exit hook.

then, the initialization of bisect_ppx runs when the top-level
definitions of `at_exit_main.ml` are evaluated: the sigterm handler
and dump at_exit hook are installed.

then the application starts. it forks and the child exits immediately.
the at_exit hooks installed now are `[bisect; application]` and they
are executed in LIFO. hence bisect's hook runs first. then the child's
hook runs. it sleeps.

meanwhile, the parent process kills the child. if the child did not
have it's default sigterm restored, then this kill would be ignored
and we should see the message "I should've been woken up by now.". 

the number of traces written are two: one from the at_exit hook in each process.

does that make sense? i'd be happy to make something simpler if you have any nice ideas.

otherwise, i don't mind if you merge it as it is, and we can iron out any wrinkles later on.